game-core/src/main/java/games/strategy/engine/framework/startup/ui/pbem/ForumPosterEditor.java

```diff
@@ -0,0 +1,201 @@
+package games.strategy.engine.framework.startup.ui.pbem;
+
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import org.triplea.java.ViewModelListener;
+import org.triplea.swing.DocumentListenerBuilder;
+import org.triplea.swing.SwingComponents;
+
+/** A class for selecting which Forum poster to use. */
+class ForumPosterEditor extends JPanel implements ViewModelListener<ForumPosterEditorViewModel> {
+  private static final long serialVersionUID = -6069315084412575053L;
+  private final JButton viewPosts = new JButton("View Forum");
+  private final JButton testForum = new JButton("Test Post");
+  private final JTextField topicIdField = new JTextField();
+  private final JLabel topicIdLabel = new JLabel("Topic Id:");
+  private final JLabel forumLabel = new JLabel("Forums:");
+  private final JCheckBox attachSaveGameToSummary = new JCheckBox("Attach save game to summary");
+  private final JCheckBox alsoPostAfterCombatMove = new JCheckBox("Also Post After Combat Move");
+  private final JComboBox<String> forums = new JComboBox<>();
+
+  ForumPosterEditor(final ForumPosterEditorViewModel viewModel) {
+    super(new GridBagLayout());
+
+    viewModel.setView(this);
```

Comment:
Most of the code in this class is original, note anywhere there is a call to `viewModel`, that will be new as well as the 'updateState' method from `implements ViewModelListener<ForumPosterEditorViewModel>`